app/vmui/packages/vmui/src/components/LineChart/LineChart.tsx

```diff
@@ -1,196 +1,127 @@
-/* eslint max-lines: ["error", {"max": 300}] */
-import React, {useCallback, useMemo, useRef, useState} from "react";
-import {line as d3Line, max as d3Max, min as d3Min, scaleLinear, ScaleOrdinal, scaleTime} from "d3";
-import "./line-chart.css";
-import Measure from "react-measure";
-import {AxisBottom} from "./AxisBottom";
-import {AxisLeft} from "./AxisLeft";
-import {DataSeries, DataValue, TimeParams} from "../../types";
-import {InteractionLine} from "./InteractionLine";
-import {InteractionArea} from "./InteractionArea";
-import {Box, Popover} from "@material-ui/core";
-import {ChartTooltip, ChartTooltipData} from "./ChartTooltip";
-import {useAppDispatch} from "../../state/common/StateContext";
-import {dateFromSeconds} from "../../utils/time";
-import {MetricCategory} from "../../hooks/useSortedCategories";
-
-interface LineChartProps {
-  series: DataSeries[];
-  timePresets: TimeParams;
-  height: number;
-  color: ScaleOrdinal<string, string>; // maps name to color hex code
-  categories: MetricCategory[];
-}
-
-interface TooltipState {
-  xCoord: number;
-  date: Date;
-  index: number;
-  leftPart: boolean;
-  activeSeries: number;
-}
-
-const TOOLTIP_MARGIN = 20;
-
-export const LineChart: React.FC<LineChartProps> = ({series, timePresets, height, color, categories}) => {
-  const [screenWidth, setScreenWidth] = useState<number>(window.innerWidth);
-
+import React, {FC, useMemo} from "react";
+import {Line} from "react-chartjs-2";
+import {Chart, ChartData, ChartOptions, ScatterDataPoint, TimeScale} from "chart.js";
+import zoomPlugin from "chartjs-plugin-zoom";
+import {getNameForMetric} from "../../utils/metric";
+import "chartjs-adapter-date-fns";
+import debounce from "lodash.debounce";
+import {TimePeriod} from "../../types";
+import {useAppDispatch, useAppState} from "../../state/common/StateContext";
+import {dateFromSeconds, getTimeperiodForDuration} from "../../utils/time";
+import {GraphViewProps} from "../Home/Views/GraphView";
+Chart.register(zoomPlugin);
```

Comment:
I am not really sure what may happen with 2 line charts on a page registering zoom Plugin... if plugins are supposed to be registered globally once - it should be moved somewhere to separate file...

from what I see in the docs this is indeed global setting https://www.chartjs.org/docs/latest/developers/plugins.html#global-plugins

so it should not be here.

other option is to use per chart plugins and options

Reply:
updated